pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -322,22 +337,92 @@ func (c *podIPMetricsCollector) setupDirectoryWatcher(dir string) error {
 	return nil
 }
 
+func (c *podIPMetricsCollector) countTotalIPs(subnet *net.IPNet) (uint64, error) {
+	ones, bits := subnet.Mask.Size()
+	if bits == 0 {
+		return 0, fmt.Errorf("invalid IP subnet mask %v", subnet.Mask)
+	}
+	if ones == bits {
+		ones = bits - 1
+	}
+	return uint64(1 << (bits - ones)), nil
+}
+
+func (c *podIPMetricsCollector) calculateAssignedIP(subnetFile string) error {
+	var assignedIPv4, assignedIPv6 uint64
+	networkConfigList, err := libcni.ConfListFromFile(subnetFile)
+	if err != nil {
+		glog.Errorf("Error reading network config from file %s: %v", subnetFile, err)
+		return err
+	}
+
+	var ipamConfig *allocator.IPAMConfig
+	for _, plugin := range networkConfigList.Plugins {
+		if plugin.Network.Type == "gke" {
```

Comment:
Ref https://github.com/GoogleCloudPlatform/netd/blob/a9375688ce84c55389fa63a16feee07906da2400/scripts/install-cni.sh#L87-L91

Will we have / when will we have `"type": "ptp"`?

Reply:
This looks like legacy logic added for the ptp->gke migration. Need to verify further though - probably on the enablement path e.g. enabling netd on an existing cluster?

Reply:
I just finished testing the enable netd on an existing cluster. I found that although we can see netd enabled, the file name is different regardless of whether it is enabled before recreating a node. Therefore, I think the path should not change before recreating and always be 'ptp'.